### PR TITLE
Add Arch Linux support

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: manifests/install.pp [hunk 1]
```diff
@@ -4,8 +4,10 @@
 class zram_generator::install (
   Enum['installed', 'absent'] $install_defaults = $zram_generator::install_defaults,
 ) {
-  package { 'zram-generator-defaults':
-    ensure => $install_defaults,
+  unless $facts['os']['name'] == 'Archlinux' {
```
Comment (ekohl) [inline manifests/install.pp:7]: I'm not a fan of OS specific conditionals in code. Would it make sense to have a `manage_defaults` parameter instead?

**Comment by hashworks** [comment]
I've tested this on an Arch Linux machine with no configured swap / zram – worked just fine, repeated `puppet apply` calls ran without error.

File: manifests/install.pp (hunk 1 above)
Comment (bastelfreak) [reply manifests/install.pp]: updated it

Comment (bastelfreak) [reply manifests/install.pp]: @ekohl thinking about it, by default ensure is set to absent for zram-generator-defaults. We could keep the code like it is on master and it works out of the box, but will fail on Arch Linux when someone sets it to installed. should we catch that and call fail()?

File: manifests/init.pp [hunk 2]
```diff
@@ -1,9 +1,11 @@
 # @summary Install and configure zram_generator zram mounts
 # @see https://github.com/systemd/zram-generator
 # @param install_defaults Controls if the package 'zram-generator-defaults' be installed.
+# @param manage_defaults_package wether the zram-generator-defaults package should be installed
 #
 class zram_generator (
   Enum['installed', 'absent'] $install_defaults = 'absent',
+  Boolean $manage_defaults_package = $facts['os'] ? { 'Archlinux' => false, default => true },
```
Comment (ekohl) [inline manifests/init.pp:8]: You can even consider not exposing it and only keep it as an internal private variable (thus only leaving it to Hiera), but at least correct the logic:
```suggestion
  Boolean $manage_defaults_package = $facts['os']['name'] != 'Archlinux',
```

**Review by ekohl — CHANGES_REQUESTED** [review]

**Review by ekohl — APPROVED** [review]

File: spec/classes/init_spec.rb [hunk 3]
```diff
@@ -16,7 +16,11 @@
         it { is_expected.to contain_exec('generate_zram_units').with_refreshonly(true) }
         it { is_expected.to contain_file('/usr/lib/systemd/zram-generator.conf.d').with_ensure('directory') }
         it { is_expected.to contain_package('zram-generator-defaults').with_ensure('absent') }
-        it { is_expected.to contain_package('zram-generator').with_ensure('installed') }
+        if facts [:os]['name'] == 'Archlinux'
+          it { is_expected.not_to contain_package('zram-generator')
+        else
+          it { is_expected.to contain_package('zram-generator').with_ensure('installed') }
```
Comment (ekohl) [inline spec/classes/init_spec.rb:22]: `zram-generator` should still be present - it's the defaults (a line above) that you need to change.

**Pull request merged**